### PR TITLE
genact 0.4.0 (new formula)

### PR DESCRIPTION
**Pull request by weihanglo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/genact.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Genact < Formula
+  desc "Nonsense activity generator"
+  homepage "https://github.com/svenstaro/genact"
+  url "https://github.com/svenstaro/genact/archive/0.4.0.tar.gz"
+  sha256 "92265af4fc7adceda4379336ec5440125b60556dca59f16de37bfb32c35c5da8"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "build", "--release"
```
Comment (commitay) [inline Formula/genact.rb]: `system "cargo", "install", "--root", prefix` and remove the `bin.install`

File: Formula/genact.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    bin.install "target/release/genact"
+  end
+
+  test do
+    system "#{bin}/genact", "-V"
```
Comment (commitay) [inline Formula/genact.rb]: The test needs to exercise and verify the functionality of the software.

e.g. `assert_match` the output from `--list-modules`?

**Review by commitay — DISMISSED** [review]

File: Formula/genact.rb (hunk 1 above)
Comment (weihanglo) [reply Formula/genact.rb]: Fixed

File: Formula/genact.rb (hunk 2 above)
Comment (weihanglo) [reply Formula/genact.rb]: Added tests.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
Thanks @weihanglo for the pull request!

**Comment by xeoneux** [comment]
@fxcoudert was #23154 rejected because it did not add the rust dependency and did not build/install it using cargo?

**Comment by ilovezfs** [comment]
@xeoneux it was rejected because we require source builds. See

https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Formulae.md#we-dont-like-binary-formulae

**Pull request closed without merging**